app/controllers/profile_controller.rb

```diff
@@ -6,5 +6,32 @@ def require_user
     render file: "/public/404" unless current_user?
   end
 
-  def index; end
+  def index
+    @user = current_user
+  end
+
+  def edit
+    @user = current_user
+  end
+
+  def update
+    @user = current_user
+    @user.update(user_params)
+    if User.where(email:@user.email).where.not(id:@user.id) != []
+      flash[:failure] = "That email already exists"
+      render :edit
+    elsif @user.save
+      redirect_to '/profile'
+      flash[:error] = "Your profile is updated"
+    else
+      flash[:failure] = "You are missing required fields."
+      render :edit
+    end
```

Comment:
User Story 20. Didn't call for validating all data was present, but we added in code similar to what was done at user creation to maintain consistency.